application/src/main/java/org/thingsboard/server/config/DeviceRpcRequestSizeFilter.java

```diff
@@ -15,31 +15,31 @@
  */
 package org.thingsboard.server.config;
 
-import com.amazonaws.HttpMethod;
 import jakarta.servlet.FilterChain;
 import jakarta.servlet.ServletException;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
-import org.springframework.security.web.util.matcher.NegatedRequestMatcher;
-import org.springframework.security.web.util.matcher.RequestMatcher;
 import org.springframework.stereotype.Component;
+import org.springframework.util.AntPathMatcher;
 import org.springframework.web.filter.OncePerRequestFilter;
 import org.thingsboard.server.common.msg.tools.TbMaxPayloadSizeExceededException;
 import org.thingsboard.server.exception.ThingsboardErrorResponseHandler;
 
 import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 @Slf4j
 @Component
 @RequiredArgsConstructor
 public class DeviceRpcRequestSizeFilter extends OncePerRequestFilter {
 
-    private final RequestMatcher uriMatcher = new AntPathRequestMatcher("/api/v1/*/rpc/**", HttpMethod.POST.name());
-    private final RequestMatcher matcher = new NegatedRequestMatcher(uriMatcher);
+    private final Set<String> urls = new HashSet<>(List.of("/api/v1/*/rpc/**", "/api/plugins/rpc/**", "/api/rpc/**"));
```

Comment:
immutable `List.of` is enough. Or use Set.of(). But you never search/modify by URL pattern. So the set makes no sense here